src/components/Schema/Property.tsx

```diff
@@ -6,55 +6,68 @@ import GithubSlugger from 'github-slugger';
 import PropTypes from 'prop-types';
 import * as React from 'react';
 
+import {
+  inferType,
+  isArray,
+  isBoolean,
+  isObject,
+  isString,
+} from '../../utils/json';
 import { formatSchemaName } from '../../utils/openapi';
-import Example from './Example';
+import Example from '../Example';
+import SchemaContext from './SchemaContext';
 import ServiceTags from './ServiceTags';
 import Tag from './Tag';
 
 import styles from './Property.module.scss';
 
 const slugger = new GithubSlugger();
 
-type Service = 'score' | 'factors' | 'insights';
-
 type TagValue = boolean | string | number | null;
-
-export type PropertyType = 'array<boolean>'
-  | 'array<number>'
-  | 'array<integer>'
-  | 'array<object>'
-  | 'array<string>'
-  | 'boolean'
-  | 'number'
-  | 'integer'
-  | 'object'
-  | 'string';
-
 export interface IProperty {
-  children?: React.ReactElement,
-  example?: string;
+  children?: React.ReactElement | React.ReactElement[],
   linkToSchemaName?: string;
   name: string;
   schemaId?: string;
-  services?: '*' | Service[];
+  services?: MinFraudServices;
   tags?: Record<string, TagValue>;
-  type: PropertyType;
+  type?: SchemaPropertyType;
 }
 
 const Property: React.FC<IProperty> = (props) => {
-  const location = useLocation();
-
   const {
     children: description,
-    example,
     linkToSchemaName,
     tags: schemaTags,
-    schemaId,
     name,
     services,
     type,
   } = props;
 
+  const location = useLocation();
+
+  const {
+    json: schemaJson,
+    id: schemaId,
+    jsonPointer: schemaJsonPath,
+    services: schemaServices,
+  } = React.useContext(SchemaContext);
+
+  const [
+    example,
+    setExample,
+  ] = React.useState('');
+
+  const jsonPointer = React.useMemo(
```

Comment:
It doesn't look like we're getting any savings with `useMemo`.  We should remove this so that we don't incurr any overhead.

Reply:
Good call. I've implemented this.